### PR TITLE
54 bug 10a gwas arr v matrix

### LINKED ISSUE #54: [Bug]: ./10a-gwas_aar.sh: Error: the V matrix is not invertible
### Contact Details

ks164@duke.edu

### Scripts

10a-gwas_aar.sh

### What happened?

Running fastGWA on twin (MZ and DZ) data gives the following error:

Estimating the genetic variance (Vg) by fastGWA-REML (grid search)...
Error: the V matrix is not invertible.

the input files data_fast.grm.sp contains 2175 pairs and data_fast.grm.id contains 1502 individuals in case it helps.

full log output below.

### How can the bug be reproduced?

_No response_

### R version

4.3.3 (February, 2024)

### Python version

None

### Relevant log output

```shell
*******************************************************************
* Genome-wide Complex Trait Analysis (GCTA)
* version v1.94.1 Linux
* Built at Nov 15 2022 21:14:25, by GCC 8.5
* (C) 2010-present, Yang Lab, Westlake University
* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>
*******************************************************************
Analysis started at 12:22:33 EDT on Mon Oct 21 2024.
Hostname: dcc-login-01

Options: 
 
--grm /hpc/group/sugdenlab/godmc_phase2/ERisk/processed_data/genetic_data/data 
--make-bK-sparse 0.05 
--autosome 
--out /hpc/group/sugdenlab/godmc_phase2/ERisk/processed_data/genetic_data/data_fast 
--thread-num 16 

The program will be running with up to 16 threads.
Pruning the GRM to a sparse matrix with a cutoff of 0.050000...
Total number of parts to be processed: 1
    Saving 1502 individual IDs
    Processing part 1
Saving the sparse GRM (2175 pairs) to [/hpc/group/sugdenlab/godmc_phase2/ERisk/processed_data/genetic_data/data_fast.grm.sp]
finished generating a sparse GRM Success:

Analysis finished at 12:22:33 EDT on Mon Oct 21 2024
Overall computational time: 0.58 sec.
Done on making bK sparse
*******************************************************************
* Genome-wide Complex Trait Analysis (GCTA)
* version v1.94.1 Linux
* Built at Nov 15 2022 21:14:25, by GCC 8.5
* (C) 2010-present, Yang Lab, Westlake University
* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>
*******************************************************************
Analysis started at 12:22:33 EDT on Mon Oct 21 2024.
Hostname: dcc-login-01

Options: 
 
--bfile /hpc/group/sugdenlab/godmc_phase2/ERisk/processed_data/genetic_data/data 
--grm-sparse /hpc/group/sugdenlab/godmc_phase2/ERisk/processed_data/genetic_data/data_fast 
--fastGWA-mlm 
--mpheno 1 
--pheno age_acc.plink 
--h2-limit 20 
--out /hpc/group/sugdenlab/godmc_phase2/ERisk/results/10/DNAmAgeSD 

Reading PLINK FAM file from [/hpc/group/sugdenlab/godmc_phase2/ERisk/processed_data/genetic_data/data.fam]...
1502 individuals to be included from FAM file.
Reading phenotype data from [age_acc.plink]...
1502 overlapping individuals with non-missing data to be included from the phenotype file.
1502 individuals to be included. 762 males, 740 females, 0 unknown.
Reading PLINK BIM file from [/hpc/group/sugdenlab/godmc_phase2/ERisk/processed_data/genetic_data/data.bim]...
7159238 SNPs to be included from BIM file(s).
Reading the sparse GRM file from [/hpc/group/sugdenlab/godmc_phase2/ERisk/processed_data/genetic_data/data_fast]...
After matching all the files, 1502 individuals to be included in the analysis.
Estimating the genetic variance (Vg) by fastGWA-REML (grid search)...
Error: the V matrix is not invertible.
An error occurs, please check the options or data
```

### PR DESCRIPTION
**Pull request by SiyiSEA**
To fix #54 , I remove the related individuals in the dataset before go through the GWAS and heritability calculation.

The commit of [remove chr-set 23 flag](https://github.com/genetics-of-dna-methylation-consortium/godmc_phase2/pull/80/commits/9340f3baa27c53fcd14d2ec6ae4d3000bed19d77) has been reverted.

### CODE REVIEW

File: 10b-heritability_aar.sh [hunk 1]
```diff
@@ -18,7 +18,7 @@ for clock_name in $clock_names
 do
   echo "Runing SNP heritability for " $clock_name $i
   ${gcta} \
-          --grm ${grmfile_all}  \
+          --grm ${grmfile_all}_gaws10  \
```
Comment (ejh243) [inline 10b-heritability_aar.sh:21]: should this be _gwas10?

File: 11c-check_compress_data.sh [hunk 2]
```diff
@@ -164,7 +171,7 @@ done
 echo ""
 echo "Compressing the outputs from 03a, 10 and 11"
 cd ${home_directory}
-tar -zcf results/AgeSmokGWAS_${study_name}.tgz results/03/age_prediction.pdf results/03/age_prediction_correlation.png results/03/age_prediction_stats.csv results/03/age_prediction_stats_corrsd.csv results/10 results/11 
+tar -zcf results/AgeSmokGWAS_${study_name}.tgz results/03/logs_a results/03/age_prediction.pdf results/03/age_prediction_correlation.png results/03/age_prediction_stats.csv results/03/age_prediction_stats_corrsd.csv results/03/smoking_prediction.pdf results/10 results/11
```
Comment (ejh243) [inline 11c-check_compress_data.sh:174]: IS results/03/logs_a a file or folder? If a folder, will this work?

**Review by ejh243 — APPROVED** [review]

File: 10b-heritability_aar.sh (hunk 1 above)
Comment (SiyiSEA) [reply 10b-heritability_aar.sh:21]: Yes, ${grmfile_all}_gaws10 filters out the related individuals based on the threshold 0.05, while ${grmfile_all} didn't.

File: 11c-check_compress_data.sh (hunk 2 above)
Comment (SiyiSEA) [reply 11c-check_compress_data.sh]: Sorry, i miss the log.txt in the end. I've fixed it.

**Pull request merged**